app/controllers/openid_connect/discovery_controller.rb

```diff
@@ -0,0 +1,51 @@
+module OpenidConnect
+  class DiscoveryController < ApplicationController
+    def show
+      case params[:id]
+        when 'webfinger'
+          webfinger_discovery
+        when 'openid-configuration'
+          openid_configuration
+        else
+          raise HttpError::NotFound
+      end
+    end
+
+    private
+
+    def webfinger_discovery
+      jrd = {
+        links: [{
+                  rel: OpenIDConnect::Discovery::Provider::Issuer::REL_VALUE,
+                  href: absoluteURL("/")
+                }]
+      }
+      jrd[:subject] = params[:resource] if params[:resource].present?
+      render json: jrd, content_type: "application/jrd+json"
+    end
+
+    def openid_configuration
+      config = OpenIDConnect::Discovery::Provider::Config::Response.new(
+        issuer: absoluteURL("/"),
+        registration_endpoint: absoluteURL(openid_connect_clients_path),
+        authorization_endpoint: absoluteURL(new_openid_connect_authorization_path),
+        token_endpoint: absoluteURL(openid_connect_access_tokens_path),
+        userinfo_endpoint: absoluteURL(api_v0_user_path),
+        jwks_uri: "https://not_configured_yet.com", # TODO: File.join({new_openid_connect_authorization_path} + "/jwks.json"),
+        scopes_supported: Scope.all.collect(&:name),
+        response_types_supported: OAuthApplication.available_response_types,
+        request_object_signing_alg_values_supported: %i(HS256 HS384 HS512),
+        subject_types_supported: %w(public pairwise),
+        id_token_signing_alg_values_supported: %i(RS256),
+        token_endpoint_auth_methods_supported: %w(client_secret_basic client_secret_post),
+        # TODO: claims_supported: ["sub", "iss", "name", "email"]
+      )
+      render json: config
+    end
+
+    def absoluteURL(url)
```

Comment:
This is a controller, you really shouldn't need this method at all. Just use the `_url` helpers instead of the `_path` helpers.